docs/cookbook/recipe_creating_a_locale_switcher.rst

```diff
@@ -0,0 +1,161 @@
+Creating a locale switcher
+==========================
+
+This is a full working example of creating a locale switcher.
+
+.. note::
+    This article assumes you are using Symfony 4, Flex, and autowiring.
+
+Prerequisites
+-------------
+
+Install following Sonata dependencies:
+
+* `Sonata Easy-Extends Bundle`_
+* `Sonata Intl Bundle`_
+
+Create ``LocaleSubscriber`` and ``UserLocaleSubscriber`` as explained `here`_.
+
+.. note::
+    Sonata Intl Bundle is optional, but will allow list of locales to be shown as list of languages.
+
+Handle locale switch
+--------------------
+
+Create a controller to handle locale switch in ``src/Controller/LocaleController.php``:
+
+.. code-block:: php
+
+    <?php
+
+    namespace App\Controller;
+
+    use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+    use Symfony\Component\Routing\Annotation\Route;
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\HttpFoundation\Session\SessionInterface;
+    use Symfony\Component\HttpFoundation\RedirectResponse;
+
+    class LocaleController extends AbstractController
+    {
+        /**
+        * @Route("/locale/{locale}", name="locale")
+        */
+        public function index(Request $request, SessionInterface $session, $locale)
```

Comment:
Is session really needed as a dependency here? Can't you do `$request->getSession()`?

Reply:
Nope, removed in last commit. Thanks for your advise.